server/src/frontend_api.rs

```diff
@@ -1,3 +1,250 @@
 
-pub fn configure_frontend_api(_cfg: &mut web::ServiceConfig) {}
+use crate::types::{EdgeJsonResult, EdgeProvider, EdgeToken};
+
+#[get("/proxy/all")]
+async fn get_frontend_features(
+    edge_token: EdgeToken,
+    features_source: web::Data<dyn EdgeProvider>,
+    context: web::Query<Context>,
+) -> EdgeJsonResult<FrontendResult> {
+    let client_features = features_source.get_client_features(edge_token);
+    let context = context.into_inner();
+
+    let evaluated_features = resolve_frontend_features(client_features, context);
+
+    Ok(Json(FrontendResult {
+        toggles: evaluated_features,
+    }))
+}
+
+#[actix_web::post("/proxy/all")]
+async fn post_frontend_features(
+    edge_token: EdgeToken,
+    features_source: web::Data<dyn EdgeProvider>,
+    context: web::Json<Context>,
+) -> EdgeJsonResult<FrontendResult> {
+    let client_features = features_source.get_client_features(edge_token);
+    let context = context.into_inner();
+
+    let evaluated_features = resolve_frontend_features(client_features, context);
+
+    Ok(Json(FrontendResult {
+        toggles: evaluated_features,
+    }))
+}
+
+fn resolve_frontend_features(
+    client_features: ClientFeatures,
+    context: Context,
+) -> Vec<EvaluatedToggle> {
+    let mut engine = EngineState::default();
+    engine.take_state(client_features.clone());
+
+    client_features
+        .features
+        .into_iter()
+        .map(|toggle| {
+            let variant = engine.get_variant(toggle.name.clone(), &context);
+            EvaluatedToggle {
+                name: toggle.name.clone(),
+                enabled: engine.is_enabled(toggle.name, &context),
+                variant: EvaluatedVariant {
+                    name: variant.name,
+                    enabled: variant.enabled,
+                    payload: variant.payload.map(|succ| Payload {
+                        payload_type: succ.payload_type,
+                        value: succ.value,
+                    }),
+                },
+                impression_data: false,
+            }
+        })
+        .collect()
+}
+
+pub fn configure_frontend_api(cfg: &mut web::ServiceConfig) {
+    cfg.service(get_frontend_features);
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use crate::types::{EdgeProvider, FeaturesProvider, TokenProvider};
+    use actix_web::{
+        http::header::ContentType,
+        test,
+        web::{self, Data},
+        App,
+    };
+    use serde_json::json;
+    use unleash_types::{
+        client_features::{ClientFeature, ClientFeatures, Constraint, Operator, Strategy},
+        frontend::{EvaluatedToggle, EvaluatedVariant, FrontendResult},
+    };
+
+    #[derive(Clone, Default)]
+    struct MockDataSource {
+        features: Option<ClientFeatures>,
+    }
+
+    impl MockDataSource {
+        fn with(self, features: ClientFeatures) -> Self {
+            return MockDataSource {
+                features: Some(features),
+            };
```

Comment:
Just curious, can this be:

```rs
MockDataSource {
    features: Some(features),
}
```

instead?

Reply:
Yes. Yes it can. Apparently I'm still in Golang mode. Surprised Clippy didn't get cranky. Nice catch, updated